### ISSUE TITLE
Placeholders in publishedIn reference

### ISSUE THREAD

**Issue by gbif-portal** [gbif/portal-feedback#142]
**Placeholders in publishedIn reference**

In several hundred GNUB &amp; ZooBank species one can find placeholders in the publishedIn literature citation such as:

[Title not known]
[Title Not Known, please research and update]
[Unspecified Publisher]
[includes page check]
Anonymous, [NONE]

-----
fbitem-dataset34a96ebe-e51c-4222-9d08-5c2043c39dec

System: Chrome 56.0.2924 / Mac OS X 10.12.3
Referer: https://demo.gbif.org/dataset/34a96ebe-e51c-4222-9d08-5c2043c39dec/taxonomy/108753211
Window size: width 1658 - height 949
[API log](http://elk.gbif.org:5601/app/kibana#/discover/UAT-Varnish-403s?_g=(refreshInterval:(display:Off,pause:!f,value:0),time:(from:'2017-03-23T11:57:32.533Z',mode:absolute,to:'2017-03-23T12:03:32.533Z'))&_a=(columns:!(request,response,clientip),filters:!(),index:'prod-varnish-*',interval:auto,query:(query_string:(analyze_wildcard:!t,query:'response:%3E499%20AND%20(request:%22%2F%2Fapi.gbif.org%22)')),sort:!('@timestamp',desc))&indexPattern=uat-varnish-*&type=histogram)
[Site log](http://elk.gbif.org:5601/app/kibana#/discover/UAT-Varnish-403s?_g=(refreshInterval:(display:Off,pause:!f,value:0),time:(from:'2017-03-23T11:57:32.533Z',mode:absolute,to:'2017-03-23T12:03:32.533Z'))&_a=(columns:!(request,response,clientip),filters:!(),index:'prod-varnish-*',interval:auto,query:(query_string:(analyze_wildcard:!t,query:'response:%3E399%20AND%20(request:%22%2F%2Fdemo.gbif.org%22)')),sort:!('@timestamp',desc))&indexPattern=uat-varnish-*&type=histogram)

**Comment by mdoering** [comment]
@kbraak @gbif/mobilization please contact Rich Pyle about this, ZooBank is a subset view on top of GNUB and shares the data.

**Comment by kbraak** [comment]
Thanks, I have now reported the issue to @deepreef over email, copied to Helpdesk.

**Comment by deepreef** [comment]
Yes, most of these are what we call "placeholder" records.  They are cases where a genus name is registered (in order to allow the registration of a species), but the publication details are not known.  I can very easily remove these records from the output; or I can change the values that are included in the output to remove the comments (i.e., make them blank).  I will discuss this with Markus in April, unless there is a need to fix it sooner.

**Comment by mdoering** [comment]
simply replacing those placeholders with an empty string should solve it!

**Comment by deepreef** [comment]
OK, I'll plan to do exactly that.

**Comment by mdoering** [comment]
There is a similar placeholder issue for years in name authorships:

> Thelia [Year?]

from http://www.gbif.org/species/127587857 and http://zoobank.org/NomenclaturalActs/A325ADB0-E9CE-4FE2-9554-FD2FB3154142

@deepreef, please remove all placeholders whereever they are from your ZooBank & GNUB dwca exports.

**Comment by deepreef** [comment]
OK, let's plan on reviewing all of this on Sunday, and I'll solve all of these at the same time.

**Comment by MattBlissett** [comment]
@mdoering, I think this is probably fixed with the new NUB.

**Comment by MattBlissett** [comment]
@mdoering, I think this is probably fixed in the new NUB. Can you confirm?

**Comment by ManonGros** [comment]
[This dataset](https://www.gbif.org/dataset/34a96ebe-e51c-4222-9d08-5c2043c39dec) is labeled as an orphan now.

**Comment by MattBlissett** [comment]
That's probably happened since the original endpoint, http://gnub.org/data/GNUB_DwC.zip, is down. The "rescue" meant copying our last successful download of that file from temporary storage to good storage.

If it comes back, we can restore the original endpoint. The details are in the registry machine tags.

**Comment by mdoering** [comment]
@ManonGros I would restore the original endpoint right away. There is not much sense in keeping the historic copy with errors around if @deepreef decided to take it down. 

@deepreef, has the endpoint simply changed to a different location and have the publication placeholders been removed?

**Comment by deepreef** [comment]
We're in the process of setting up new servers (FINALLY -- 18 months after we got the budget approved!  Don't ask!).  We should have the new servers fully functional before the end of this month.  At that time, I will create new endpoints for ZooBank (which you have been using), full GNUB (which this thread is about), and also for our Bishop Museum Occurrence data (which is currently hosted by VertNet).  When I'm ready to implement this, I'll get in touch with you directly, @mdoering , for a little bit of "hand holding".

**Comment by ManonGros** [comment]
@deepreef how are the new servers?
Have you had the opportunity to create new endpoints for ZooBank, full GNUB, and for your Bishop Museum Occurrence data?
Do you need any help with updating the endpoints on GBIF?

**Comment by deepreef** [comment]
@ManonGros  Sorry for the delayed reply -- I've been travelling almost nonstop these past two months.

So... it has now been over THREE YEARS since we received budget approval for our new web servers, and although the servers themselves were finally set up a little over a year ago, we've only managed to get them fully operational within the last couple of months (!!!).  It's been a long, sad story... BUT!!  The good news is that I believe all barriers have now been removed.  I created a test IPT at http://ipt.bishopmuseum.org:8080/ipt/ and it seems to be working OK.  I'm ready to replace this with a permanent/published version, but I didn't want to do that until we were ready to make the transition from the existing host (VertNET) to our own local host.  Ideally, I would like to maintain our existing GBIF datasetid (b929f23d-290f-4e85-8f17-764c55b3b284), and just change the host to point to our own IPT server (after I create the live/published version).  I am on my way to Berlin, where I will be seeing @mdoering -- so maybe this will be the best opportunity to make the change?

**Comment by ManonGros** [comment]
Thank @deepreef for your reply!
The procedure to migrate a dataset from one IPT to another without generating duplicates is the following:
Once the new dataset is in the new IPT, before clicking on register:
* click on “edit” Metadata,
* then click on “Additional Metadata”,
* scroll at the bottom on the page and click on “Add new alternative identifier”
enter the GBIF dataset identifier (it is the UUID at the end of the GBIF URL to your dataset).

If the dataset has the same title and publishing organization as in the old IPT, the endpoint will be automatically updated.

**Comment by jlegind** [comment]
@deepreef Has this taxonomic resource been fixed yet?

**Comment by deepreef** [comment]
FINALLY got the servers back online on Monday this week (exactly 6 months to the day after we lost them to Ransomware attack).  I'm spending this week getting them configured, and hope to have IPT installed and online by the end of next week, or first week of June.

The good news is that the Museum is investing in a 5-year, $5M Department of Informatics, starting with hiring a Director of Informatics in the next few months. This means that I will no longer be the bottleneck in developing and maintaining our IPT resources.

We also just completed a MASSIVE hardware/infrastructure upgrade, which will help in myriad ways.

**Comment by jlegind** [comment]
@deepreef The Global Names Usage Bank dataset has been orphaned. This takes place when a dataset url has been offline or otherwise unreachable for an extended period of time.
This wiki page offers an in depth explanation: https://github.com/gbif/watchdog/wiki/OrphanedDatasets#ways-to-diagnose-if-a-dataset-has-been-orphaned-or-not 

I was hoping that GLA would adopt the resource and keep it actively curated.

**Comment by deepreef** [comment]
The database is very active, but it recently moved servers and we’ve been working with our IT folks (VERY frustrating experience!) to clear up some firewall issues in order to get the IPT up and running again.  I will make another attempt this week to sort out the issue so that the IPT is functional.  As soon as I have an update, I will let you know.
